src/nodes/lighting/EnvironmentNode.js

```diff
@@ -71,14 +71,11 @@ class EnvironmentNode extends LightingNode {
 
 		//
 
-		const envMap = material.envMap;
-		const intensity = envMap ? reference( 'envMapIntensity', 'float', builder.material ) : reference( 'environmentIntensity', 'float', builder.scene ); // @TODO: Add materialEnvIntensity in MaterialNode
```

Comment:
When adding `materialEnvRotation` to `MaterialNode`, I fixed this TODO and also implemented `materialEnvIntensity`.

Reply:
I moved to `Material Properties`. If we can use `uniform()` with `onRender Update()` it will be better than using `references()`.